### PR TITLE
Implements phone auth support for Admin node.js SDK.

### PR DESCRIPTION
**Pull request by bojeil-google**
Adds phoneNumber to UserRecord and UserInfo.
Implements new API for looking up users by phone number: getUserByPhoneNumber
Adds the ability to pass the phoneNumber property when creating a new user,
or updating an existing one via createUser and updateUser.
This includes the ability to delete the phone number for a specified user.
Defines the new related auth errors:
auth/invalid-phone-number and auth/phone-number-already-exists.

### CODE REVIEW

File: test/unit/auth/auth-api-request.spec.ts [hunk 1]
```diff
@@ -534,6 +564,75 @@ describe('FirebaseAuthRequestHandler', () => {
...
+  describe('getAccountInfoByPhoneNumber', () => {
+    const httpMethod = 'POST';
+    const host = 'www.googleapis.com';
+    const port = 443;
+    const path = '/identitytoolkit/v3/relyingparty/getAccountInfo';
+    const timeout = 10000;
+    it('should be fulfilled given a valid phoneNumber', () => {
+      const expectedResult = {
+        users : [
+          {phoneNumber: '+11234567890'},
```
Comment (hiranya911) [inline test/unit/auth/auth-api-request.spec.ts]: Should we also add the `providerUserInfo` field to the expected response, and see if get parsed correctly?

**Comment by hiranya911** [comment]
I believe we also need to update `index.d.ts` right?

File: test/unit/auth/auth-api-request.spec.ts (hunk 1 above)
Comment (bojeil-google) [inline test/unit/auth/auth-api-request.spec.ts]: I added a more realistic response and I test the expected response is returned from this operation which is enough for this API. This API returns a response that resolves with the same server response. UserRecord parses that content. I have tests for that in that file. auth.ts passes the response to UserRecord. I have a check for the whole flow in auth.spec.ts. I stub the response of getAccountInfoByPhoneNumber and then check that it is parsed correctly into the expected UserRecord.

**Comment by bojeil-google** [comment]
Thanks for catching the missing definitions in index.d.ts. I added those. I also added detailed response for getAccountInfoByPhoneNumber. More detailed checks on parsing is adding for getUserByPhoneNumber.

**Comment by hiranya911** [comment]
LGTM

File: test/integration/auth.js [hunk 2]
```diff
@@ -33,23 +33,75 @@ function test(utils) {
...
+    // Delete any existing users that could affect the test outcome.
+    var promises = [
+      deletePhoneNumberUser(testPhoneNumber),
+      deletePhoneNumberUser(testPhoneNumber2),
+      deletePhoneNumberUser(nonexistentPhoneNumber),
+      deletePhoneNumberUser(updatedPhone)
+    ];
+    return Promise.all(promises)
+      .catch(function(error) {
+        utils.logFailure('setUp()', error);
```
Comment (hiranya911) [inline test/integration/auth.js]: Perhaps log this as `before()` to be clear?

**Comment by hiranya911** [comment]
LGTM with one minor suggestion.

File: test/integration/auth.js (hunk 2 above)
Comment (bojeil-google) [inline test/integration/auth.js]: Thanks for catching this mislabel. I originally had named this setUp() but then renamed it to before().

**Pull request merged**